src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -41,14 +42,22 @@ namespace Windows.UI.Xaml.Media
 {
 	public partial class VisualTreeHelper
 	{
-		private static readonly List<WeakReference<IPopup>> _openPopups = new List<WeakReference<IPopup>>();
+		private static readonly WeakReferenceEqualityComparer<IPopup> _openPopupsEqualityComparer = new();
+		private static readonly List<WeakReference<IPopup>> _openPopups = new();
 
 		internal static IDisposable RegisterOpenPopup(IPopup popup)
 		{
-			var weakPopup = new WeakReference<IPopup>(popup);
+			var popupRegistration = _openPopups.FirstOrDefault(
+				p => WeakReferenceExtensions.GetTarget(p) == popup);
 
-			_openPopups.AddDistinct(weakPopup);
-			return Disposable.Create(() => _openPopups.Remove(weakPopup));
+			if (popupRegistration is null)
+			{
+				popupRegistration = new WeakReference<IPopup>(popup);
```

Comment:
It'll be best if you use ManagedWeakReference and the WeakReferencePool, similar to this: https://github.com/unoplatform/uno/blob/84c97baf5c2340e8b1e4014b4eb8d1b8f83dc951/src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs#L55